tests/test_errors.py

```diff
@@ -0,0 +1,147 @@
+import warnings
+
+import pytest
+import torch
+
+import pytorch_finufft
+
+torch.manual_seed(0)
+# devices
+
+
+def test_t1_mismatch_cuda_non_cuda() -> None:
+    points = torch.rand((2, 10), dtype=torch.float64)
+    values = torch.randn(10, dtype=torch.complex128).to("cuda:0")
+
+    with pytest.raises(ValueError, match="Some tensors are not on the same device"):
+        pytorch_finufft.functional.finufft_type1.apply(points, values, (10, 10))
+
+
+def test_t1_mismatch_cuda_index() -> None:
+    points = torch.rand((2, 10), dtype=torch.float64).to("cuda:0")
+    values = torch.randn(10, dtype=torch.complex128).to("cuda:1")
```

Comment:
this would be a test case that requires 2 gpus then? ~Could also do `"cpu"` and `"cuda"` devices?~
Can we find a way of naming this so it's easy to exclude?

Reply:
Ah, that case is tested above, OK

Reply:
I made the name contain `multi_cuda`, so you could do `pytest -k "cuda and not multi"` on a 1-gpu machine